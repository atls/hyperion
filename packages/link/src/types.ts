export interface NextLinkProps {
  fontFamily?: string
  fontWeight?: string | number | string[] | number[]
  lineHeight?: string | number | string[] | number[]
  fontSize?: string | number | string[] | number[]
  color?: string
  hoverColor?: string
  clickedColor?: string
  href: string
  target?: string
  display?: string
  as?: string
  query?: any
  textTransform?: string
  textAlign?: string
  underline?: boolean
  width?: string
  whiteSpace?: string
  dangerouslySetInnerHTML?: any
  textDecorationStyle?: string
  textDecorationColor?: string
  passHref?: boolean
  children?: any
}

export interface LinkProps {
  children?: any
  href?: string
  fontWeight?: string | number | string[] | number[]
  lineHeight?: string | number | string[] | number[]
  fontSize?: string | number | string[] | number[]
  fontFamily?: string
  whiteSpace?: string
  dangerouslySetInnerHTML?: any
  color?: string
  hoverColor?: string
  textDecorationStyle?: string
  textDecorationColor?: string
  clickedColor?: string
  textTransform?: string
  underline?: boolean
  target?: string
  width?: string
  pt?: string
  onClick?: any
  onMouseOver?: () => void
  onMouseLeave?: () => void
}
