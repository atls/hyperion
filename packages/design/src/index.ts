export * from './Sample'
export * from './Layout'
export * from './Container'
export * from './List'
export * from './Text'
export * from './Space'
