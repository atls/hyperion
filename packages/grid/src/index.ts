export * from './Column'
export * from './Grid'
