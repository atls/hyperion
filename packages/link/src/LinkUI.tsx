import { layout } from 'styled-system'

import styled     from '@emotion/styled'
import { Text }   from '@atlantis-lab/text'

export const LinkUI = styled<any>(Text.withComponent('a'))(
  {
    position: 'relative',
    display: 'inline-flex',
  },
  layout,
)
