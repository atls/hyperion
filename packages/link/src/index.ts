export * from './Link'
export * from './NextLink'
export * from './RouteLink'
