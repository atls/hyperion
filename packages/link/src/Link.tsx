/* eslint-disable react/require-default-props */

import React, { FC, useState } from 'react'

import { LinkUI }              from './LinkUI'
import { LinkProps }           from './types'

export const Link: FC<LinkProps> = React.forwardRef(
  (
    { children, href, fontSize, lineHeight, fontWeight, color, hoverColor, clickedColor, ...props },
    ref,
  ) => {
    const [hovered, setHovered] = useState(false)
    const [clicked, setClicked] = useState(false)
    return (
      <LinkUI
        fontSize={fontSize}
        lineHeight={lineHeight}
        ref={ref}
        fontWeight={fontWeight}
        /* eslint-disable no-nested-ternary */
        color={clicked ? clickedColor || color : hovered ? hoverColor || color : color}
        href={href}
        onMouseOver={() => setHovered(true)}
        onFocus={() => setHovered(true)}
        onMouseLeave={() => {
          setHovered(false)
          setClicked(false)
        }}
        onMouseDown={() => setClicked(true)}
        onMouseUp={() => setClicked(false)}
        {...props}
      >
        {children}
      </LinkUI>
    )
  },
)
