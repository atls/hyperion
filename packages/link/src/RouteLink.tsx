import React               from 'react'
import { Link, LinkProps } from 'react-router-dom'

import { useTheme }        from '@emotion/react'

import { LinkUI }          from './LinkUI'

const RouteLinkStyled = LinkUI.withComponent(Link)

export const RouteLink: LinkProps = (props) => {
  const theme = useTheme()

  return <RouteLinkStyled theme={theme} {...props} />
}
