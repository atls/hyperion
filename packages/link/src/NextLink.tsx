/* eslint-disable react/require-default-props */

import Link              from 'next/link'
import React             from 'react'

import ThemedLinkUI      from './LinkUI'
import { NextLinkProps } from './types'

export const NextLink = ({
  href,
  children,
  as,
  query,
  passHref = true,
  ...props
}: NextLinkProps) => (
  <Link href={{ pathname: href, query: query || {} }} as={as || href} passHref={passHref}>
    <ThemedLinkUI {...props}>{children}</ThemedLinkUI>
  </Link>
)

NextLink.defaultProps = {
  fontFamily: 'primary',
  fontWeight: '',
}
