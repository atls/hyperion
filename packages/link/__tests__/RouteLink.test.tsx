import React             from 'react'
import renderer          from 'react-test-renderer'
import { MemoryRouter }  from 'react-router-dom'

import { Text }          from '@atlantis-lab/text'
import { ThemeProvider } from '@emotion/react'

import { RouteLink }     from '../src'

describe('test suit for link', () => {
  it('should render Link correctly', () => {
    const defaultTheme = {
      colors: {
        white: '#FFF',
        lightBlue: '#00F',
      },
    }

    const tree = renderer
      .create(
        <MemoryRouter>
          <ThemeProvider theme={defaultTheme}>
            <RouteLink to='/nowhere'>
              <Text color='green' fontSize='60px' fontWeight={700}>
                Nowhere
              </Text>
            </RouteLink>
          </ThemeProvider>
          ,
        </MemoryRouter>,
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
