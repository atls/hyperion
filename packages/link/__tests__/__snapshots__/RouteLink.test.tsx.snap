// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suit for link should render Link correctly 1`] = `
Array [
  .emotion-1 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-0 {
  color: green;
  font-size: 60px;
  font-weight: 700;
  font-family: primary;
}

<a
    className="emotion-1"
    href="/nowhere"
    onClick={[Function]}
  >
    <span
      className="emotion-0"
      color="green"
      fontFamily="primary"
      fontSize="60px"
      fontWeight={700}
    >
      Nowhere
    </span>
  </a>,
  ",",
]
`;
